src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -515,7 +515,7 @@ private DType(DKind kind, string namedValueKind)
             AssertValid();
         }
 
-        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null)
+        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null, HashSet<IExternalTabularDataSource> associatedDataSources = null)
```

Comment:
>HashSet<IExternalTabularDataSource> associatedDataSources  [](http://example.com/codeflow?start=112&length=58)

This is little odd pattern that you can externally associate datasources to a type? Why do we need to it this way?